iree/compiler/Dialect/VMLA/IR/VMLABase.td

```diff
@@ -160,19 +160,25 @@ class VMLA_ElementTypeOp<string mnemonic, list<OpTrait> traits = []> :
   }];
 }
 
-class VMLA_UnaryOp<string mnemonic, Attr typeAttr, list<OpTrait> traits = []> :
-    VMLA_ElementTypeOp<mnemonic, traits> {
+class VMLA_UnaryOp<string mnemonic, Attr typeAttr, list<OpTrait> traits = [
```

Comment:
If you mark VMLA_Buffer as a buildable type, you should be able to leave off `!vmla.buffer` (the dst type) entirely.

Add another superclass to VMLA_Buffer (sorry I'd do a suggested edit, but GitHub doesn't let you comment on unchanged lines.

`BuildableType<"::mlir::iree_compiler::IREE::VMLA::BufferType::get($_builder.getContext())">`

This should also allow you to drop the AllTypesMatch constraint here, which doesn't do any work for the verifier since VMLA_Buffer is already a concrete type.

Reply:
Oh or you can add a builderCall field to the type declaration, that's probably easier, and consistent with https://sourcegraph.com/github.com/google/iree/-/blob/iree/compiler/Dialect/IREE/IR/IREEBase.td

`let builderCall = "$_builder.getType<IREE::VMLA::BufferType>()";`